b.go

```diff
@@ -0,0 +1,158 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+
+	dependencyparse "infra/dependency_parse/kitex_gen/bytedance/bits/dependency_parse"
+	git_server "infra/dependency_parse/kitex_gen/bytedance/bits/git_server"
+	metaService "infra/dependency_parse/kitex_gen/bytedance/bits/meta"
+	"infra/dependency_parse/pkg/business"
+	constDefine "infra/dependency_parse/pkg/consts"
+	"infra/dependency_parse/pkg/service/gitserver"
+	"infra/dependency_parse/pkg/service/jenkins"
+	"infra/dependency_parse/pkg/service/meta"
+
+	"code.byted.org/bits/common_lib/consts"
+	"code.byted.org/bits/common_lib/utils"
+	"code.byted.org/gopkg/facility/fjson"
+	"code.byted.org/gopkg/logs"
+)
+
+func ParseDependency(ctx context.Context, req *dependencyparse.DependencyParseRequest) (*dependencyparse.DependencyParseResponse, error) {
+
+	err := checkCreateDevTaskParam(ctx, req)
+	if err != nil {
+		logs.CtxDebug(ctx, "error %+v", err)
+		return nil, err
+	}
+
+	app, err := meta.QueryAppInfo(ctx, req.UniqueId)
+	if err != nil {
+		logs.CtxDebug(ctx, "error %+v", err)
+		return nil, err
+	}
+
+	project, err := getProjectInfoByURL(ctx, app.GitUrl)
+	if err != nil {
+		logs.CtxDebug(ctx, "error %+v", err)
+		return nil, err
+	}
+
+	ref := req.GetRef()
+	if ref == "" {
+		ref = project.DefaultBranch
+	}
+
+	components := make([]*dependencyparse.ComponentInfo, 0)
+	targets := make([]string, 0)
+
+	if app.TechnologyStack == metaService.TechnologyStack_iOS {
+		components, targets, err = business.ParseIosDeps(ctx, app, project, ref)
+	} else if app.TechnologyStack == metaService.TechnologyStack_Android {
+		components, targets, err = business.ParseAndroidRegisteredDeps(ctx, int64(project.Id), ref)
+	} else if app.TechnologyStack == metaService.TechnologyStack_Flutter {
+		components, targets, err = business.ParseFlutterYamlDeps(ctx, int64(project.Id), ref)
+	} else {
+		err = fmt.Errorf("unsupported tech stack %v", app.TechnologyStack)
+	}
+
+	if err != nil {
+		logs.CtxDebug(ctx, "error %+v", err)
+		return nil, err
```

Comment:
[](url)```suggestion
		components, targets, err = business.ParseFlutterYamlDeps(ctx, int64(project.Id), ref)
	} else {
		err = fmt.Errorf("unsupported tech stack %v", app.TechnologyStack)
	}

	if err != nil {
		logs.CtxDebug(ctx, "error %+v", err)
		return nil, err
```
[123]([url](https://github.com/hashicorp/consul/pulls))

Reply:
[33333]([url](https://github.com/hashicorp/consul/pulls))